xarray/backends/h5netcdf_.py

```diff
@@ -164,6 +181,7 @@ def open(
             "invalid_netcdf": invalid_netcdf,
             "decode_vlen_strings": decode_vlen_strings,
             "driver": driver,
+            "storage_options": storage_options,
```

Comment:
If we can obtain the `fsspec` first, we could simplify this significantly along those lines:

```python
# get open fsspec-handle first
if storage_options is not None:
    filename = _find_absolute_paths(filename, engine="h5netcdf", backend_kwargs=dict(storage_options=storage_options))

# other code

manager = CachingFileManager(
            h5netcdf.File, filename, mode=mode, kwargs=kwargs
        )
```

`_find_absolute_paths` would need changes to cover for this, though.